Code Review

Wocked61 opened the pull request:
Group members, please make comments on the code in this area It has all the pushes made from the beginning, I believe, I chose this way to do it because it was easier to do and more convenient

Andrew-K-M committed:
made board and can move pieces

Andrew-K-M committed:
added capture pieces

Andrew-K-M committed:
changed piece selection

Andrew-K-M committed:
added change to king and double hop

AzaanMP committed:
Create index.html

AzaanMP committed:
Create script.js

AzaanMP committed:
Create style.css

AzaanMP committed:
Update index.html

AzaanMP committed:
Update script.js

AzaanMP committed:
Update style.css

AzaanMP committed:
Update script.js

Andrew-K-M committed:
refactor take piece

Andrew-K-M committed:
Merge branch 'main' into frontend

Andrew-K-M committed:
Merge pull request #2 from Wocked61/frontend

Frontend

eposri committed:
removed node_modules folder

eposri committed:
fixing linters

Andrew-K-M committed:
fixed moveSet cords

Andrew-K-M committed:
Merge branch 'main' of github.com:Wocked61/cpsc_362_sp2025_group2

Andrew-K-M committed:
fixed in bounds check for king

Andrew-K-M committed:
changed starting player

Wocked61 committed:
fixed the backend files, added my desgin to the fronend and merged the main code from andrew's efforts into the whole design

Wocked61 committed:
added some request for the javascript and a border for easier visibility for the user on the red on red constrast

Wocked61 committed:
fixed the footer alignment real quick

Andrew-K-M committed:
added a check for outof bounds move

AzaanMP committed:
Add files via upload

AzaanMP committed:
Add files via upload

AzaanMP committed:
Added Highlight Moveable Piece feature

Currently broken where onlt the 4 front pieces of the black side is highlighted only

AzaanMP committed:
Updated CSS for highlight feature

Wocked61 committed:
merged the main code to see the updates

Wocked61 committed:
fixed the cursor pointer on the board/pieces

Wocked61 committed:
added logic for the settigns and css will add the html later

Wocked61 committed:
added the popup and forgot to mention set a function to the new game function

Wocked61 committed:
Merge branch 'main' into Fontend_design

Wocked61 committed:
Merge pull request #5 from Wocked61/Fontend_design

Fontend design

Wocked61 committed:
fixed the board border

Wocked61 committed:
Merge branch 'Fontend_design' of https://github.com/Wocked61/cpsc_362_sp2025_group2 into Fontend_design

Wocked61 committed:
Fixed the timer to not start until a move has been made

Wocked61 committed:
added instructions to the help button and fixed the timer width

Wocked61 committed:
fixed a bug in the colour changer where if the bottom was a different colour than black when changing the top colour the bottom colour would go back to black

Wocked61 committed:
i think i can fix the checkmark by adding this in the pything file
also i added the help button functionality

Wocked61 committed:
made a function so the python files run

Wocked61 committed:
made some fixes to the settings and the new game logic theres still some bugs to fix

Wocked61 committed:
Merge pull request #6 from Wocked61/Fontend_design

Fontend design bracnh stuff

Wocked61 committed:
added more pages for the website for filler right now

Wocked61 committed:
Merge pull request #7 from Wocked61/Fontend_design

added more pages for the website for filler right now

Wocked61 committed:
made some formatting fixes to merge

Wocked61 committed:
trying again for the merging linter fix

Wocked61 committed:
Merge pull request #9 from Wocked61/Fontend_design

Fontend design testing

eposri committed:
fixing black.yml and prettier.yml files

eposri committed:
black linter auto changes

eposri committed:
testing black.yml and prettier.yml files

eposri committed:
trying to get prettier to run action

eposri committed:
prettier linter auto changes

eposri committed:
linter merged into 1 file due to pushing twice conflicts, testing

eposri committed:
linters auto changes

eposri committed:
testing if action works when prettier and black both make formatting changes

eposri committed:
linters auto changes

eposri committed:
created .gitignore to fix node_modules

Wocked61 committed:
quick patch to where the checkers would freeze when loading the page

Wocked61 committed:
Merge branch 'main' into Fontend_design

Wocked61 committed:
Merge pull request #10 from Wocked61/Fontend_design

quick patch to where the checkers would freeze when loading the page

Wocked61 committed:
linters auto changes

sonykiim committed:
playing around w the sound

AzaanMP committed:
Highlight feature for pieces added

AzaanMP committed:
linters auto changes

sonykiim committed:
playing around w the sound

sonykiim committed:
linters auto changes

Wocked61 committed:
added the code for the sound for when we move the pieces and the click the buttons

Wocked61 committed:
linters auto changes

Wocked61 committed:
fixed the highlighting for possible pieces

Wocked61 committed:
linters auto changes

Wocked61 committed:
added some sounds and fixed sounds

Wocked61 committed:
linters auto changes

Wocked61 committed:
Merge pull request #11 from Wocked61/main

merging to my branch so i can push the code without pushing to main

Wocked61 committed:
made the chess.js file need to fix the chess board to appear with the css its easier to do at home

Wocked61 committed:
fixed the colors for kings when changing color settings

Wocked61 committed:
adjusted the values for easier visibility

Wocked61 committed:
fixed the players turn in the display information area

Wocked61 committed:
added a promote sound for kings

Wocked61 committed:
adding changes i made before going to bed but i added a end game sound

Wocked61 committed:
updated the home page for improved design for the boxes

Wocked61 committed:
removed the top nav bar

Wocked61 committed:
fixed the chess files and will add more features later

Wocked61 committed:
added the navigation for the chess game in the home menu

Wocked61 committed:
added the chess timer and fixed the board

Wocked61 committed:
fixed some visuals with the piece border and errors in the console when the game is running which is the audio

Wocked61 committed:
added a endgame noise and popup based on the timer or who takes all the pieces

Wocked61 committed:
fix a typo in the updateScore function where it caused player 2 to get double the intended poitns

Wocked61 committed:
added a drag and drop feature need to clean up the cursor icons im going to sleep

Wocked61 committed:
improved the sounds

AzaanMP committed:
Added GameEnds Files so it checks if the game ends

AzaanMP committed:
Add files via upload

AzaanMP committed:
linters auto changes

Wocked61 committed:
changed the cursor for drag and drop to grab

Wocked61 committed:
Merge pull request #12 from Wocked61/Fontend_design

Fontend design
with features completed

Wocked61 committed:
linters auto changes

Wocked61 committed:
move variables to the top for consistentcy

Wocked61 commented on src/script.js:
```diff
@@ -0,0 +1,139 @@
+}
+
+
+setupBoard();
```
This was fixed in future pushes but it should be in a function that can be called when the game starts

Wocked61 reviewed (commented):
everything looks good and solid great work besides that one line of code which works anyways but for future implementations

Wocked61 reviewed (commented):
looks good you fixed your issues/logic and typos in comments

Wocked61 reviewed (commented):
looks good added more functions for the pieces on the board

Wocked61 reviewed (commented):
looks great king functionality and double hop was added

Wocked61 reviewed (commented):
this is just a file being made nothing to review

Wocked61 reviewed (commented):
This is also just a file being made nothing to review

Wocked61 reviewed (commented):
This is also just a file being made nothing to review

Wocked61 reviewed (commented):
This is also just a file being made nothing to review

Wocked61 reviewed (commented):
this is solid but we dont need to make individual files for the backend if its a javascript and we already have some established with andrew's code

Wocked61 reviewed (commented):
This is also just a styling being made nothing to review besides it look good

Wocked61 reviewed (commented):
this is just 1 deleted line you pushed

Wocked61 reviewed (commented):
looks good added functions refined logic its solid

Wocked61 reviewed (commented):
this is a merging of code into the main branch so I can do my design and add it later looks good

Wocked61 reviewed (commented):
merging of my branch to the main from what I can see

Wocked61 reviewed (commented):
looks good this is just deleting files

Wocked61 reviewed (commented):
deleting a path from the formatter looks good simple 2 lines

Wocked61 reviewed (commented):
moving code around to the top for consistency

Wocked61 reviewed (commented):
deleting code and using node?

Wocked61 reviewed (commented):
fixing logic of the king piece looks good

Wocked61 reviewed (commented):
changed the starting player looks good nothing much to review

Wocked61 reviewed (commented):
deleted frontend files from the backend because it shouldn't be in there when we have established files

Wocked61 reviewed (commented):
this was after the presentation and i fixed the footer because it wasn't aligned correctly

Wocked61 reviewed (commented):
looks good fixing of logic for the king movement

Wocked61 reviewed (commented):
this is good adding the timer should have a push comment related to what was added but your role that you chose is backend

Wocked61 reviewed (commented):
this is good adding the timer logic should have a push comment related to what was added

Wocked61 reviewed (commented):
looks good highlights pieces the bottom code was fixed later to reduce redundancy

Wocked61 reviewed (commented):
looks good just some css styling

Wocked61 reviewed (commented):
this is me merging the code from the main so I can add to it in my branch

Wocked61 reviewed (commented):
fixed the pointer for the pieces it was too big and in the wrong area

Wocked61 reviewed (commented):
added functionality to the settings in the HTML and javascript added the function for new game

Wocked61 reviewed (commented):
made the popup appear when the settings would be opened

Wocked61 reviewed (commented):
merging of code to release the updates of the functions to main

Wocked61 reviewed (commented):
there was an issue with the linter I was testing and trying to figure out the issue when pusing

Wocked61 reviewed (commented):
added a border to the board and moved the startup functions which need to be in a start game function for redundancy

Wocked61 reviewed (commented):
merging of code to also test for the linter

Wocked61 reviewed (commented):
fixed the timer to not start until the game is start not when the page is loaded

Wocked61 reviewed (commented):
added a help button to show how to play checkers

Wocked61 reviewed (commented):
fixing the change colour of the pieces would be tied together and reset one side if the bottom colour was changed

Wocked61 reviewed (commented):
testing for linter it was still broken and I had to add something to be able to push

Wocked61 reviewed (commented):
still testing for linter errors

Wocked61 reviewed (commented):
added a score board and fixed some game logic

Wocked61 reviewed (commented):
merging of code nothing much to review

Wocked61 reviewed (commented):
started the home and chess page to make multiple pages to navigate

Wocked61 reviewed (commented):
merging of code to main to display to other teammates

Wocked61 reviewed (commented):
fixed formatting of the code in new files

Wocked61 reviewed (commented):
removed everything from the python file to test for the linter again

Wocked61 reviewed (commented):
merging again to test the linter almost fixed it by this time

Wocked61 reviewed (commented):
fixing of the linter looks good

Wocked61 reviewed (commented):
auto format

Wocked61 reviewed (commented):
renaming of commit message

Wocked61 reviewed (commented):
testing for auto format

Wocked61 reviewed (commented):
auto formating working

Wocked61 reviewed (commented):
fixing of linter and focusing of one linter instead of two

Wocked61 reviewed (commented):
auto changes

Wocked61 reviewed (commented):
testing for linter auto format

Wocked61 reviewed (commented):
deleting prettier files

Wocked61 reviewed (commented):
fixing of a bug when loading the page

Wocked61 reviewed (commented):
merging of code to quick patch the main code

Wocked61 reviewed (commented):
highlight code to show how to highlight pieces

Wocked61 reviewed (commented):
adding sounds to the game

Wocked61 reviewed (commented):
deleting comments

Wocked61 reviewed (commented):
adding more sounds should have the sounds in a function for redundancy

Wocked61 reviewed (commented):
fixed the highlighting for smoother transition

Wocked61 reviewed (commented):
added some sounds and more variables should put the sounds uptop and make them in a function for redundancy

Wocked61 reviewed (commented):
merging of code before the review

Wocked61 reviewed (commented):
adding visuals to the chess page and adding the logic from checkers to base chess

Wocked61 reviewed (commented):
fixed the kings colours when changing colours in the settings to where it would be the same

Wocked61 reviewed (commented):
changed the values of the colours

Wocked61 reviewed (commented):
fixing the display of colours in the info area

Wocked61 reviewed (commented):
promote sound, the sounds should be in a function to reduce redundancy

Wocked61 reviewed (commented):
end game sound added need end game popup to be added but sounds should be in a function to reduce redundancy

Wocked61 reviewed (commented):
added home page changes and navigation instead of a nav bar two squares can be selected

Wocked61 reviewed (commented):
removed top nav bar

Wocked61 reviewed (commented):
fixed the chess display and backend

Wocked61 reviewed (commented):
added chess option to the home page

Wocked61 reviewed (commented):
timer for the chess page added

Wocked61 reviewed (commented):
fixed errors in the console and visuals of the pieces

Wocked61 reviewed (commented):
end game popup to be used with the sound

Wocked61 reviewed (commented):
fixed an error in player score value

Wocked61 reviewed (commented):
drag and drop feature added

Wocked61 reviewed (commented):
fixed sounds and deleted old files

Wocked61 reviewed (commented):
end game files were made but the end game popup was already added this shouldn't be In a new file in the backend files

Wocked61 reviewed (commented):
pushing of code

Wocked61 reviewed (commented):
fixed the cursor icon when drag and dropping

Wocked61 reviewed (commented):
last merging of code for code review

AzaanMP commented:
coes look great except for a couple of problems to be fixed

AzaanMP commented:
King piece double capture fails on 8th rank end move.

Samsondu commented:
Review of getValidMoves() function
Reviewer: Samson Du

Purpose of function: return an array of valid moves for a single piece

Positives:
- consistent naming convention 
- easy to digest code
- No major issues noted 

Code suggestions: 
- instead of hard coding 1 and -1 in the moveset, use variable names 
- Could also combine checking if the piece is a king into one if statement instead of checking twice since kings all have the same moveset

Wocked61 committed:
updated the to do list for future implementation

kevinchau1 commented:
Code Review - Kevin Chau

gameends.js

Whitebox Testing
--------------
createpiece(total) -- works correctly to create a list of ' { alive: true} ' objects.
Maybe in the future add piece attributes like board position to expand the game features?
checkgameover() -- works correctly, tested with both sides

Blackbox Testing
----------------
Testing individual functions
- createpiece(5) , works as intended creates 5 pieces.
- new CheckersGame(red, black, 300, 300) , checkers object is created with correct properties

Structure of code is very neat and organized.

Wocked61 committed:
fixed the timer by making it wider to not have the asymetrical visually apauling

Andrew-K-M committed:
fixed for king double hop to the side

Samsondu committed:
Update script.js

Wocked61 committed:
pulling idk what i changed here

Wocked61 committed:
Merge branch 'Fontend_design' of https://github.com/Wocked61/cpsc_362_sp2025_group2 into Fontend_design

Wocked61 committed:
added to do list changes

Wocked61 committed:
Merge branch 'Fontend_design' into Highlight-function-fix

Wocked61 committed:
Merge pull request #22 from Wocked61/Highlight-function-fix

Highlight function fix

Wocked61 committed:
Merge branch 'Fontend_design' into frontend

Wocked61 committed:
Merge pull request #23 from Wocked61/frontend

fixed the conflicts and merging to my branch

Wocked61 committed:
fixed the end game popup centering, king visual acessibility with visual feedback, deleted files not being used

Wocked61 committed:
fixed bugs in the endgame popup timer the time would keep going and when closing the end game popup if you moved a piece the time would keep going so I disabled the ability to move pieces after the game ends

Wocked61 committed:
add a endgame popup when there are no valid moves

Wocked61 committed:
fixed a quick bug where it would display the wrong player that won when theres no valid moves

Wocked61 committed:
fixed the button sound for the settings

Wocked61 committed:
fixed the highlight pieces function and added a show valid moves feature

Wocked61 committed:
changed the valid moves function to display a dot instead of changing the square colour

Wocked61 committed:
added the ability to change the colours fo the board

Wocked61 committed:
put the sounds in a function to make it easier to call and change

Wocked61 committed:
added a setting to disable sounds and the chess logic will work on the front end later

Wocked61 committed:
got the chess game to work need to spend time making it look nice and add quality of life features

Wocked61 committed:
changed the board design will add more later

kevinchau1 committed:
Added chess images for later finished implementation of chess logic

Wocked61 committed:
improved the visual clarity of the pieces

Wocked61 committed:
added the ability to see where pieces can go and attack other pieces

Wocked61 committed:
added the red border to the check feature instead of an alert

Wocked61 committed:
added sounds for a responsive feel

Wocked61 committed:
added a drag and drop feature to the chess pieces

removed unnecessary files

and added a picture to the home page for chess

Wocked61 committed:
added Sony's features and fleshed them out

Wocked61 committed:
fix two major bugs where the wins would increment by 2 or 3 and when promoting the player that promoted can move again

kevinchau1 committed:
implemented a color picker for chess, maybe will add color picker for pieces later

kevinchau1 committed:
added template for chess piece theme, only classic for now

Wocked61 committed:
set up the divs for the chess page

Wocked61 committed:
Merge branch 'Fontend_design' of https://github.com/Wocked61/cpsc_362_sp2025_group2 into Fontend_design

Wocked61 committed:
added Kevins feature of the colors and fleshed them out with the updated divs

Wocked61 committed:
minor issues fixed such as en passant not playing a sound and drag and drop not making the pieces look correct

Wocked61 committed:
fixed the error in the console

kevinchau1 committed:
removed typing to promotion mechanic, now added a GUI for more easier interactibility

Wocked61 committed:
Merge pull request #24 from Wocked61/new-promotion-branch

removed typing to promotion mechanic, now added a GUI for more easier…

kevinchau1 committed:
Added capture images to chess boxes

Wocked61 committed:
Merge pull request #26 from Wocked61/new-promotion-branch

Added capture images to chess boxes

Wocked61 committed:
fixed kevins function of displaying pieces taken

Wocked61 committed:
fixed a bug where pressing new game would disable the function of the captured pieces

sonykiim committed:
updated design

Wocked61 committed:
removing reset win records button

Wocked61 committed:
added a endGame popup
fixed the game info board
Fixed the settings
fixed a button on the bottom

Wocked61 committed:
fixed the timer and added a popup for when the time runsout with the audio

fixed  the timer in the settings

formatted all the other code
removed useless files

sonykiim committed:
updated look

kevinchau1 committed:
Added a new primary font to chess ui

Wocked61 committed:
added display pieces thats been taken

Wocked61 committed:
Merge branch 'Fontend_design' of https://github.com/Wocked61/cpsc_362_sp2025_group2 into Fontend_design

Wocked61 committed:
fixed footer for homepage and comments on scriptjs

Wocked61 committed:
updates before the presentation

Samsondu committed:
Update script.js

Bug fix for get valid moves, also made it recursive when there are hops

Samsondu committed:
Update script.js

deleted console log comments

Samsondu committed:
Merge branch 'Fontend_design' of https://github.com/Wocked61/cpsc_362_sp2025_group2 into Fontend_design

Samsondu committed:
fixed merge issues and deleted console logs

Wocked61 committed:
Merge pull request #28 from Wocked61/Fontend_design

Fontend design

Wocked61 committed:
linters auto changes

Wocked61 committed:
Update README.md

added more details to the readme

Wocked61 committed:
linters auto changes
